lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -981,7 +1000,15 @@ def payment_details(payment_method, options = {})
         when String
           token_type_and_details(payment_method)
         else
-          type = network_token?(payment_method) || options[:wallet_type] == :google_pay ? :network_token : :credit
+          type = if network_token?(payment_method) || options[:wallet_type] == :google_pay
```

Comment:
Is there a way to not have nested if statements here? Can we put another case for `NetworkTokenizationCreditCard`?

Reply:
Yeah let me fix this

Reply:
I'll update it a bit but I can't change it to NetworkTokenizationCreditCard because network_token?(payment_method) was added by Shopify and I don't want to break an integration